### PR TITLE
refactor: use new IPFS async/await APIs

### PR DESCRIPTION
**Pull request by alanshaw**
Use the new IPFS async/await APIs.

Also upgrades `ipfsd-ctl` dep and test setup.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,115 +1,92 @@
...
-    readableStream.pipe(responseStream)
-
-    return new Promise((resolve, reject) => {
-      readableStream.once('error', (error) => {
-        if (error) {
-          log(error)
-          return resolve(new Response(error.toString(), header(500, 'Error fetching the file')))
+    if (typeof Blob === 'undefined') {
+      const responseStream = toStream.readable((async function * () {
+        for await (const chunk of source) {
```
[the remark this answers is not in the document]
Comment (alanshaw) [reply src/index.js]: Oh nice I didn't know that existed.

**Comment by alanshaw** [comment]
@vasco-santos this is another one where the tests probably shouldn't depend on IPFS. The new js-ipfs depends on this change so this module needs to be released before we can release js-ipfs.

File: src/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (alanshaw) [reply src/index.js]: I added it to https://github.com/alanshaw/it-awesome

**Comment by vasco-santos** [comment]
Agree! I will open an issue to refactor the tests

**Comment by alanshaw** [comment]
@vasco-santos if it wasn't obvious, this is a breaking change!

**Comment by vasco-santos** [comment]
Yep, otherwise `js-ipfs` breaks. 

It should be in the commit! but I will add it on merge

**Comment by vasco-santos** [comment]
`0.5.0` is ready 🚢

**Pull request merged**